File: c/PROBLEMS/1557.c
#include <stdio.h>
#include <math.h>
int qtdalgarismoo (int n);
void matriz(int n);
void matriz(int n){
    int i, j, x;
    x=0;
    int big = qtdalgarismoo(pow(2,(2*n-2)));

    for(i=0;i<n;i++){
        for (j=0;j<n;j++){
            if (j == 0){
                printf("%*d", big, (int) pow(2, x+j));

            }else{
                printf(" %*d", big, (int) pow(2, x+j));
            }

        }
        printf("\n");
        x+=1;
    }
    printf("\n");
}

int qtdalgarismoo (int n){
    int digitos=0;
    do
    {

        n % 10;
        n = n/10;
        digitos++;

    } while (n > 0);
    return digitos;
}

int main(){
    while (1){
        int n;
        scanf("%d", &n);
        if (n != 0){
            matriz(n);
        }else{
            break;
        }
    }
}

File: js/PROBLEMS/1557.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8')
var lines = input.split('\n')
lines = lines.map(x => parseInt(x))

function qtdalgarismoo (n){
    return String(n).length
}

function matriz(n) {
    let i, j
    let x = 0
    big  = qtdalgarismoo(Math.pow(2, (2*n-2)))

    for (i=0;i<n;i++){
        s = ''
        for(j=0;j<n;j++){
            if (j === 0) {
                s += String((Math.pow(2, x + j))).padStart(big);
            } else {
                s += ' ' + String((Math.pow(2, x + j))).padStart(big);
            }
            
        }
        console.log(s)
        x++
    }
    console.log()
}

while (1){
    n = lines.shift()
    if (n!=0){
        matriz(n)
    }else{
        break
    }
}